### PR TITLE
#50 observable type gets removed

### PR DESCRIPTION
**Pull request by ZakariaeMkassi**
If we declare a property without assigning it a specific type (e.g. number ,any ...), its value will not be read and therefore not merged since the compiler tries to use the value before giving it a specific value is assigned.

Therefore, if we enter simple parameter types without curly brackets {} in Observable<>, it will be merged.

### CODE REVIEW

File: test/curly_brackers_test.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import merge from "../src/index";
+import { expect } from "chai";
+import "mocha";
+
+describe("merging property without initialization its value", () => {
```
Comment (Zylesto) [inline test/curly_brackers_test.ts:5]: Merging a property without initializing its value,

File: test/curly_brackers_test.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    };
+    return this.http.post<{ content: SampledataModel[] }>(
+      this.urlService + 'search',
+      searchCriteria,
+    );
+  }}
+  `,
+        patch = "";
+
+  it("Test", () => {
```
Comment (Zylesto) [inline test/curly_brackers_test.ts:30]: does not yield a proper observable type.

File: test/curly_brackers_test.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    const result: String = merge(base, patch, true).replace(/\n/g, " ");
+    expect(result).equal(
+      "import Observable from 'rxjs/observable';   class test { getSampleData(size: number, page: number): Observable<{}> {  const searchCriteria: SearchCriteria = { pageable: { pageSize: size, pageNumber: page }, name:  searchTerms.name, surname:  searchTerms.surname };      return this.http.post<{ content: SampledataModel[] }>(       this.urlService + 'search',       searchCriteria,     ); }   } "
+    );
+  });
+});
+
+describe("merging without curly brackets {}", () => {
```
Comment (Zylesto) [inline test/curly_brackers_test.ts:38]: Merging it with proper syntax,

File: test/curly_brackers_test.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    };
+    return this.http.post<{ content: SampledataModel[] }>(
+      this.urlService + 'search',
+      searchCriteria,
+    );
+  }}
+  `,
+        patch = "";
+
+  it("Test", () => {
```
Comment (Zylesto) [inline test/curly_brackers_test.ts:63]: yields a proper observable type

File: test/curly_brackers_test.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+describe("Declaration curly brackets {} with type inizialisation", () => {
+  const base = `
+  const searchCriteria: any = {
+    name: 'test',
+    surname: 'test',
+  };
+  `,
+        patch = "";
+
+  it("Test", () => {
```
Comment (Zylesto) [inline test/curly_brackers_test.ts:81]: yields a proper merged declaration.

File: test/curly_brackers_test.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      " const searchCriteria: any = { name: 'test', surname: 'test' }; "
+    );
+  });
+});
+
+// if we declare a property without assigning it to a specific type (e.g. number ,any ...) 
+//its value will not be read and thus not merged, because the compiler tries to use the value 
+//before it is assigned a specific value.
+
+describe("Declaration curly brackets {} without type inizialisation", () => {
```
Comment (Zylesto) [inline test/curly_brackers_test.ts:93]: Declaring a variable without proper syntax,

File: test/curly_brackers_test.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  const base = `
+const searchCriteria: {
+    name: 'test',
+    surname: 'test',
+  };
+  `,
+        patch = "";
+
+  it("Test", () => {
```
Comment (Zylesto) [inline test/curly_brackers_test.ts:102]: does not yield a valid declaration.

**Review by Zylesto — CHANGES_REQUESTED** [review]
Maybe you should leave out tests which are not working and instead proof, that if you use correct syntax for the observable type, the merge will work. because if you run the tests, they will always fail and this indicates future developers that something might be broken, although it isn't.

**Comment by ZakariaeMkassi** [comment]
> Maybe you should leave out tests which are not working and instead proof, that if you use correct syntax for the observable type, the merge will work. because if you run the tests, they will always fail and this indicates future developers that something might be broken, although it isn't.


All tests that have been written already work, otherwise the pipeline will fail.

File: test/curly_brackers_test.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        pageNumber: page,
+      },
+      name: searchTerms.name,
+      surname: searchTerms.surname,
+    };
+    return this.http.post<{ content: SampledataModel[] }>(
+      this.urlService + 'search',
+      searchCriteria,
+    );
+  }}
```
Comment (EduardKrieger) [inline test/curly_brackers_test.ts:26]: Is all of this needed for the test?

Comment (EduardKrieger) [reply test/curly_brackers_test.ts:26]: same in the other test cases

**Pull request closed without merging**